lab_2_keywords_cooccurrence/main.py

```diff
@@ -55,7 +99,17 @@ def calculate_word_degrees(candidate_keyword_phrases: KeyPhrases,
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not check_list(candidate_keyword_phrases, tuple, False) or not  check_list(content_words, str, False):
+        return None
+    dict_degree = {}
+    for phrase in candidate_keyword_phrases:
+        for word in content_words:
+            if word in phrase:
+                dict_degree[word] = len(phrase) + dict_degree.get(word, 0)
+            if word not in dict_degree:
```

Comment:
can we avoing checking this condition?

Reply:
do you mean "elif" instead of it?

Reply:
not exactly 
it is just that this way you are adding this key to the dictionary multiple times 
consider adding them all at once at the very beginning and then just updating